app/src/app/utils/events/mapEvents.ts

```diff
@@ -0,0 +1,175 @@
+/**
+ Port over from map events declared at: https://github.com/uchicago-dsi/districtr-components/blob/2e8f9e5657b9f0fd2419b6f3258efd74ae310f32/src/Districtr/Districtr.tsx#L230
+ */
+import type { Map, MapLayerMouseEvent, MapLayerTouchEvent } from "maplibre-gl";
+import { useMapStore } from "@/app/store/mapStore";
+import { MutableRefObject, useRef } from "react";
+import { BLOCK_LAYER_ID } from "@/app/constants/layers";
+import { boxAroundPoint } from "../helpers";
+import React from "react";
+import { HighlightFeature, SelectFeatures } from "./handlers";
+
+/*
+MapEvent handling; these functions are called by the event listeners in the MapComponent
+*/
+
+/**
+ * What happens when the map is clicked on; incomplete implementation
+ * @param e - MapLayerMouseEvent | MapLayerTouchEvent, the event object
+ * @param map - MutableRefObject<Map | null>, the maplibre map instance
+ * @param hoverFeatureIds - React.MutableRefObject<Set<string>>, used to keep track of geoids that have been hovered over
+ */
+export const handleMapClick = (
+  e: MapLayerMouseEvent | MapLayerTouchEvent,
+  map: MutableRefObject<Map | null>,
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
+) => {
+  const mapStore = useMapStore.getState();
+  const activeTool = mapStore.activeTool;
+
+  if (activeTool === "brush" || activeTool === "eraser") {
+    const bbox = boxAroundPoint(e, mapStore.brushSize);
+
+    const selectedFeatures = map.current?.queryRenderedFeatures(bbox, {
+      layers: [BLOCK_LAYER_ID],
+    });
+
+    if (activeTool === "brush") {
+      SelectFeatures(selectedFeatures, map, mapStore);
+    } else if (activeTool === "eraser") {
+      // erase features
+      // TODO: implement eraser
+    }
+  } else {
+    // tbd, for pan mode - is there an info mode on click?
+  }
+};
+
+export const handleMapMouseUp = (
+  e: MapLayerMouseEvent | MapLayerTouchEvent,
+  map: MutableRefObject<Map | null>,
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
+) => {
+  const mapStore = useMapStore.getState();
+  const activeTool = mapStore.activeTool;
+  const isPainting = mapStore.isPainting;
+
+  if (activeTool === "brush" && isPainting) {
+    // set isPainting to false
+    mapStore.setIsPainting(false);
+  }
+};
+
+export const handleMapMouseDown = (
+  e: MapLayerMouseEvent | MapLayerTouchEvent,
+  map: MutableRefObject<Map | null>,
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
+) => {
+  const mapStore = useMapStore.getState();
+  const activeTool = mapStore.activeTool;
+
+  if (activeTool === "pan") {
+    return;
+  } else if (activeTool === "brush" || activeTool === "eraser") {
+    // disable drag pan
+    map.current?.dragPan.disable();
+    if (activeTool === "brush") {
+      mapStore.setIsPainting(true);
+      return;
+    } else if (activeTool === "eraser") {
+      // erase features tbd
+    }
+  }
+};
+
+export const handleMapMouseEnter = (
```

Comment:
These first four are just unimplemented for the time being right?

Reply:
exactly, right now these are just placeholders